source/common/local_info/local_info_impl.h

```diff
@@ -15,6 +17,12 @@ class LocalInfoImpl : public LocalInfo {
                 absl::string_view zone_name, absl::string_view cluster_name,
                 absl::string_view node_name)
       : node_(node), address_(address) {
+    // Whenever node is used to compare with wire contents, we don't want any
+    // original type information information. Also, node is likely never
+    // appearing in config dump or places where we need to recover the exact
+    // original. TODO(htuch): If we do need this at some point, add a
+    // nodeForDisplay() method.
+    Config::VersionConverter::eraseOriginalTypeInformation(node_);
```

Comment:
This type of special case scares me. Are there others like this? Is there any way to handle this more programatically?

Reply:
Eliminated, this can be eliminated due to the refactoring of how wire level representations are achieved and also due to a switch to using `LocalityTuple` for hashing in the next update.

Reply:
Eliminated.